src/utils/usePathWithoutHash.ts

```diff
@@ -0,0 +1,10 @@
+import { useRouter } from 'next/router';
+
+export function usePathWithoutHash() {
+  const router = useRouter();
+
+  // Get everything before the hashmark (#)
+  const path = router.asPath.split('#')[0];
```

Comment:
This is a string, right? So we don't need to memoize it?

Reply:
Yea `asPath` is a string